src/reflect/scala/reflect/api/TypeTags.scala

```diff
@@ -290,7 +291,7 @@ trait TypeTags { self: Universe =>
 
     def apply[T](mirror1: scala.reflect.api.Mirror[self.type], tpec1: TypeCreator): TypeTag[T] = {
       (mirror1: AnyRef) match {
-        case m: scala.reflect.runtime.JavaMirrors#MirrorImpl
+        case m: scala.reflect.runtime.JavaMirrors#JavaMirror @nowarn("cat=deprecation")
```

Comment:
Type projection causes `dealias` revealing the deprecated type underneath.
The same happens if we use `scala.reflect.runtime.JavaMirrors#JavaMirror` outside of a pattern.